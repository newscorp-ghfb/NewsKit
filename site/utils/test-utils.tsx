/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme: any = renderToFragmentWithThemeFactory(
  docsThemeLight,
);

export const mockNewsKitComponents = (...components: string[]) => () => ({
  ...(jest.requireActual('newskit') as any),
  ...components.reduce((acc, comp) => {
    acc[comp] = ({children, ...props}: any) => (
      <div
        data-comp={`NewsKit ${comp}`}
        data-testid={comp}
        data-props={JSON.stringify(props, null, 2)}
      >
        {children}
      </div>
    );
    return acc;
  }, {} as Record<string, Function>),
});
