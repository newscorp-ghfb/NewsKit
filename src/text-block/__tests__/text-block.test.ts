import {renderToFragmentWithTheme} from '../../test/test-utils';
import {TextBlock, TextBlockProps} from '..';

describe('TextBlock', () => {
  test('renders with default styles', () => {
    const fragment = renderToFragmentWithTheme(TextBlock);
    expect(fragment).toMatchSnapshot();
  });

  const tags: Array<
    ['h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'div' | 'span']
  > = [['h2'], ['span']];
  test.each(tags)('renders as %s html tag', as => {
    const fragment = renderToFragmentWithTheme(TextBlock, {
      as,
    } as TextBlockProps);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with custom style preset', () => {
    const fragment = renderToFragmentWithTheme(TextBlock, {
      stylePreset: 'inkInverse',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with custom typography preset', () => {
    const fragment = renderToFragmentWithTheme(TextBlock, {
      typographyPreset: 'editorialParagraph010',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with no text cropping', () => {
    const fragment = renderToFragmentWithTheme(TextBlock, {
      typographyPreset: 'editorialParagraph010',
      noCrop: true,
    });
    expect(fragment).toMatchSnapshot();
  });
  test('renders with logical props', () => {
    const fragment = renderToFragmentWithTheme(TextBlock, {
      paddingBlock: '30px',
      marginBlock: '30px',
    });
    expect(fragment).toMatchSnapshot();
  });
});
