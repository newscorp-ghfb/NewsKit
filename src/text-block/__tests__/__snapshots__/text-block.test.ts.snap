// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBlock renders as h2 html tag 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
}

<h2
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`TextBlock renders as span html tag 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  display: inline-block;
}

<span
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`TextBlock renders with custom style preset 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: #3358CC;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0 svg {
  fill: #3358CC;
}

.emotion-0:hover:not(:disabled) {
  color: #2B4AAB;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:hover:not(:disabled) svg {
  fill: #2B4AAB;
}

.emotion-0:active:not(:disabled) {
  color: #213A82;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:active:not(:disabled) svg {
  fill: #213A82;
}

.emotion-0:visited:not(:disabled) {
  color: #5E44E4;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:visited:not(:disabled) svg {
  fill: #5E44E4;
}

.emotion-0:visited:hover:not(:disabled) {
  color: #5E44E4;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:visited:hover:not(:disabled) svg {
  fill: #5E44E4;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`TextBlock renders with custom typography preset 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 400;
  letter-spacing: -0.5px;
  padding: 0.5px 0px;
}

.emotion-0::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-0::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`TextBlock renders with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`TextBlock renders with no text cropping 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 1.5;
  font-weight: 400;
  letter-spacing: -0.5px;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;
